### ISSUE TITLE
Getting "LS2J Error: Threw java.lang.NoClassDefFoundError: org.apache.commons.io.input.BoundedInputStream" if RunOnServer

### ISSUE THREAD

**Issue by mSIESTE** [dpastov/excel-apache-ls#2]
Getting Error:
"LS2J Error: Threw java.lang.NoClassDefFoundError: org.apache.commons.io.input.BoundedInputStream" 
if I run Agent with RunOnServer.
Can you help ?

**Comment by mSIESTE** [comment]
Full STackTrace:

"20.11.2024 13:57:47 => LS2J Error: Threw java.lang.NoClassDefFoundError: org.apache.commons.io.input.BoundedInputStream"
"20.11.2024 13:57:47 => java.lang.NoClassDefFoundError: org.apache.commons.io.input.BoundedInputStream"
"20.11.2024 13:57:47 => java.lang.NoClassDefFoundError: org.apache.commons.io.input.BoundedInputStream
	at org.apache.poi.poifs.filesystem.FileMagic.valueOf(FileMagic.java:209)
	at org.apache.poi.ss.usermodel.WorkbookFactory.create(WorkbookFactory.java:222)
	at org.apache.poi.ss.usermodel.WorkbookFactory.create(WorkbookFactory.java:185)
	at Excelreader.readExcel(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at lotus.domino.JavaConnectInvoker.invoke(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at lotus.domino.JavaConnectLoader.invoke(Unknown Source)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.input.BoundedInputStream
	at lotus.domino.JavaConnectLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClassHelper(ClassLoader.java:953)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:898)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:881)
	at lotus.domino.JavaConnectLoader.loadClass(Unknown Source)
	... 14 more
"

**Comment by dpastov** [comment]
Does it work locally?
What version of Notes/Domino do you use? I could see that in v14 they have implemented way more strict rules in terms of Java security.
Also try to see if things look fine from designer point of view.

**Comment by mSIESTE** [comment]
Yeah, locally it runs without Error.
Im using Notes v12.
I started from Designer Agent...

**Comment by dpastov** [comment]
Try to recompile please on server, it used to work on server :-)

**Comment by mSIESTE** [comment]
Recompile doesnt change anything.

**Comment by dpastov** [comment]
@mSIESTE I think I have an idea what might not working properly. The demo.nsf uses Apache POI to build excel files, while Domino 12 (perhaps v11 as well ) arrives with POI library in initial setup (it was not like that when I built this demo).

Could you check you jvm\lib\ext folder on your Domino server if any apache poi jar there?

**Comment by mSIESTE** [comment]
Sorry for the late reply.
It's been working since last week.
You're right, it was due to the missing libraries in jvm/lib/ext...
I had to copy these 6 over:

- commons-collections4-4.4.jar
- poi-4.1.1.jar
- poi-ooxml-4.1.1.jar
- poi-ooxml-schemas-4.1.1.jar
- poi-scratchpad-4.1.1.jar
- xmlbeans-3.1.0.jar

Maybe I could have saved myself the poi-scratchpad-4.1.1.jar, but I simply copied all poi* libraries to the server.

Since then everything has been going great.
